common/isolationgroup/default-isolation-group-state.go

```diff
@@ -54,17 +54,18 @@ type defaultIsolationGroupStateHandler struct {
 
 func NewDefaultIsolationGroupStateWatcher(
 	logger log.Logger,
-	domainCache cache.DomainCache,
 	config Config,
-	globalIsolationGroupDrains persistence.ConfigStore,
-	done chan bool,
+	domainCache cache.DomainCache,
+	manager persistence.ConfigStoreManager,
+	done chan struct{},
```

Comment:
why do we need an external channel?

Reply:
This is the service shutdown channel, used by the daemon to stop polling and not leak goroutines I guess? It's just a pattern used elsewhere I was following